Feature/example: Add M5Unified compatible example sketch (v0.0.3)

ito55 opened the pull request:
---

### Update: Ready for Review (2026-04-03)

This PR is now out of draft. The M5Unified example has been verified on:
* M5Stack Core2, CoreS3, Basic, M5StickC Plus

Additionally, redundant `@version` and `@date` tags have been removed from source headers for better maintenance.

---

This PR introduces a new example sketch to demonstrate the use of the Unit Encoder with the M5Unified library. This is intended to offer a modern and unified way for users to integrate the Unit Encoder across different M5Stack host devices.

### Status and Request for Collaboration

* **Status:** The new example has been confirmed working on an **M5Stack Core2** (tested by the author).
* **Request:** Since this example utilizes M5Unified for cross-platform compatibility, I'd like to confirm its stability on at least one other M5Unified-supported host device before moving this PR out of draft status and requesting final review.

**I kindly request the community to help confirm the operation on one other M5Unified host (e.g., Basic, M5StickC Plus, CoreS3, M5Paper, etc.).**

Your confirmation (or bug report) in the comments would be greatly appreciated.

### Changes Included

* New example sketch for M5Unified integration.
* Version bump to v0.0.3 in `library.properties` and `library.json`.

---

### Quick way to get the code (for Testers)

To test this branch locally, please follow the steps appropriate for your environment.

1.  `git clone https://github.com/ito55/M5Unit-Encoder`
2.  `cd M5Unit-Encoder`
3.  `git switch feature/example-m5unified`

ito55 committed:
init

copied from Unit_Encoder_M5Core2
rename
cleanup comment

ito55 committed:
Replaced the M5Core2.h library with M5Unified.h

ito55 committed:
Move Serial.println(encoder_value)

Place Serial.println(encoder_value) inside the conditional block (if statement) to print the value only when a change occurs.
This prevents continuous output and prints the encoder value only when it has been updated.

ito55 committed:
modified canvas width

Fix: Display cutoff for values over 4 digits.

ito55 committed:
config: Update metadata for M5Unified example release (v0.0.3)

ito55 committed:
feat: Improve display flexibility and cross-device compatibility

Replaced canvas.drawString with canvas.println/print for display output. This removes fixed coordinate dependencies, allowing for automatic text wrapping and easier porting to M5 controllers with different screen sizes.

ssknaoya commented on examples/Unit_Encoder_M5Unified/Unit_Encoder_M5Unified.ino:
```diff
@@ -0,0 +1,44 @@
+    cfg.clear_display = true;
+    cfg.serial_baudrate = 115200;
+    M5.begin(cfg);
+    sensor.begin();
```
When connecting the ATOMS3 and STM32F030F4P6 using the included cable, you may need to explicitly specify the pins.
```c
    // G2=SDA, G1=SCL
    Wire.begin(2, 1);
    sensor.begin(&Wire);
```

ito55 replied:
@ssknaoya, thank you for testing the operation with ATOMS3 and for sharing this valuable information. Since I don't own an ATOMS3, your cooperation is extremely helpful!

The fact that the I2C port had to be explicitly specified for ATOMS3 to function correctly suggests that we need to improve how the I2C port is designated.

While specifying `Wire.begin(2, 1);` might be the simplest solution for just the ATOMS3, as a sample sketch for a library, the ideal approach is to have common source code that works across different M5Unified controllers without hardware-specific port number designation.

I will look into ways to allow the current combination of M5Unified and the Unit Encoder library to operate without requiring explicit I2C port number specification.

Thank you again for the feedback!

ito55 committed:
feat(example): Improve I2C initialization compatibility

Enhance the sample sketch to improve compatibility across M5Unified devices, specifically addressing the I2C initialization behavior on ATOMS3.
By ensuring the proper I2C context is available before `sensor.begin()`, the example now runs reliably on ATOMS3 and maintains compatibility with other controllers without device-specific pin declarations.

ito55 commented:
@ssknaoya, I've implemented a fix based on your valuable feedback regarding the I2C initialization issue on ATOMS3. Thank you again for bringing that to my attention!
I believe this update resolves the need for explicit pin designation. Could you please confirm the operation of the Unit Encoder on your ATOMS3 using the latest sample sketch in this Pull Request?
The relevant commit is: ~[a0eb37b](https://github.com/m5stack/M5Unit-Encoder/pull/7/commits/a0eb37bdb6a0f4ba807219cfdfe304e8f9cb2c8c)~ [f282b60](https://github.com/m5stack/M5Unit-Encoder/pull/7/commits/f282b60405a11b4035c11333941f56fd176c73e7)
Your verification would be highly appreciated before merging this change.

ito55 committed:
refactor: simplify I2C initialization and remove explicit casts for readability

ito55 committed:
refactor: Simplify I2C initialization for Unit_Encoder

Move Wire.begin() into the conditional block to correctly initialize either external or default I2C pins.

Simplify the sensor.begin() call to only take the Wire instance, removing redundant pin parameters. This improves code clarity and maintains device compatibility.

ito55 committed:
Merge branch 'master' into feature/example-m5unified

ito55 committed:
fix(i2c): Remove duplicated I2C initialization

ito55 committed:
Docs: Remove version/date from source file's doc comment

This commit removes the @version and @date tags from the comment block at the top of the source file (`Unit_Encoder.h`).

The @date tag was already outdated. While the @version tag is still accurate for now, it is prone to becoming stale in future releases, creating a potential for conflicting information.

To ensure versioning is handled consistently through the official library metadata files (`library.properties` and `library.json`), these redundant lines are being proactively removed. This simplifies maintenance and prevents future confusion.

ito55 commented:
## Summary
This PR is now ready for final review. Following the initial draft, the new M5Unified-based example has been thoroughly tested across multiple devices. Additionally, I have removed redundant version and date tags from the source headers to improve maintainability.

## Changes
* **New Example:** Added a sketch demonstrating `Unit Encoder` integration with the `M5Unified` library for better cross-platform compatibility.
* **Metadata Cleanup:** Removed `@version` and `@date` tags from `Unit_Encoder.h`. 
    * *Reasoning:* These tags were often outdated or prone to becoming stale. Versioning is now strictly managed via `library.properties` and `library.json` to ensure a single source of truth and simplify future maintenance.
* **Version Bump:** Updated version to `v0.0.3` in the library metadata files.

## Verification Results
The new example sketch has been confirmed to work on the following M5Unified host devices:
* M5Stack Core2 (v1.1)
* M5Stack CoreS3-SE
* M5Stack Basic
* M5StickC Plus

Thank you to everyone who assisted with the cross-device testing!
